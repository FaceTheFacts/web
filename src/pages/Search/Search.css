.search-page {
	background: #181924;
	height: 100vh;
	width: 100vw;
	overflow: hidden;
}

.search-content {
	position: fixed;
	top: 30%;
}

.search-logo {
	width: 174.13px;
	height: 52px;
	margin-left: auto;
	margin-right: auto;
}

.search-bar {
	--background: rgba(255, 255, 255, 0.1);
	--border-radius: 8px;
	--clear-button-color: #ffffff;
	--color: #ffffff;
	--icon-color: rgba(255, 255, 255, 0.7);
	--placeholder-color: rgba(255, 255, 255, 0.5);
	--placeholder-font-style: 'inter';
	--placeholder-font-weight: 400;
	padding: 0px !important;
	width: calc(100vw - 24px);
	margin-left: 12px;
	margin-right: 12px;
}

.search-bar .searchbar-input-container {
	height: 50px;
}

@media screen and (min-width: 360px) {
	.search-bar {
		width: calc(100vw - 32px);
		margin: 12px 16px;
	}
}

@media screen and (max-width: 320px) {
	.searchbar-input.sc-ion-searchbar-md {
		font-size: 14px;
	}
}

@media screen and (max-width: 280px) {
	.searchbar-input.sc-ion-searchbar-md {
		font-size: 10px;
	}
}
