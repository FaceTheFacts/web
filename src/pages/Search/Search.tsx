import { IonContent, IonImg, IonPage, IonSearchbar } from '@ionic/react';
import React, { useState } from 'react';
import './Search.css';
import logo2021 from '../../assets/images/logo2021.svg';
import SearchScreen from './SearchScreen/SearchScreen';

const Search: React.FC = () => {
	const [showScreen, setShowScreen] = useState<boolean>(false);
	return (
		<IonPage>
			{showScreen === false ? (
				<IonContent className="search-page" fullscreen>
					<div className="search-content">
						<IonImg
							className="search-logo"
							id="logo2021"
							src={logo2021}
							alt="logo2021"
						/>
						<IonSearchbar
							className="search-bar"
							clearIcon="close-sharp"
							placeholder="Nach Kandidat:innen suchen"
							onIonFocus={(): void => setShowScreen(true)}
						></IonSearchbar>
					</div>
				</IonContent>
			) : null}

			{showScreen ? <SearchScreen /> : null}
		</IonPage>
	);
};

export default Search;
