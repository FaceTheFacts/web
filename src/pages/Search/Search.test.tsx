import React from 'react';
import { render, unmountComponentAtNode } from 'react-dom';
import { Router } from 'react-router-dom';
import { createMemoryHistory } from 'history';
import Enzyme, { mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

import Search from './Search';

let container: HTMLDivElement | null = null;

beforeEach(() => {
	// setup a DOM element as a render target
	container = document.createElement('div');
	document.body.appendChild(container);
});

afterEach(() => {
	// cleanup on exiting
	if (container !== null) {
		unmountComponentAtNode(container);
		container.remove();
		container = null;
	}
});

Enzyme.configure({ adapter: new Adapter() });

describe('integration test', () => {
	it('renders page correctly', () => {
		// given
		const history = createMemoryHistory({
			initialEntries: ['/search'],
		});

		// when
		render(
			<Router history={history}>
				<Search />
			</Router>,
			container
		);

		// then

		if (container !== null) {
			expect(container).toMatchSnapshot();
		}
	});

	it('opens camera view', () => {
		// given
		const history = createMemoryHistory({
			initialEntries: ['/search'],
		});

		// when
		const page = mount(
			<Router history={history}>
				<Search />
			</Router>
		);
		//Node.js 15 issue
		//https://solaceinfotech.com/blog/whats-new-in-node-js-15/
		process.on('unhandledRejection', () => {
			page.find('.camera-icon-background').simulate('click');
			// then
			expect(page.find('.camera-container').length).toEqual(1);
		});
	});
});
