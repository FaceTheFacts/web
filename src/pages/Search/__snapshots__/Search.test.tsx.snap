// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`integration test renders page correctly 1`] = `
<div>
  <div
    class="ion-page"
  >
    <ion-content
      class="search-page"
      fullscreen="true"
    >
      <div
        class="search-content"
      >
        <ion-img
          alt="logo2021"
          class="search-logo"
          id="logo2021"
          src="logo2021.svg"
        />
        <ion-searchbar
          class="search-bar"
          clear-icon="close-sharp"
          placeholder="Nach Kandidat:innen suchen"
        />
      </div>
    </ion-content>
  </div>
</div>
`;
